packages/cpp-debug/src/browser/memory-provider/memory-provider.ts

```diff
@@ -77,21 +70,31 @@ export class MemoryProviderImpl implements MemoryProvider {
      * @param length the length.
      * @param offset the offset.
      */
-    async readMemory(location: string, length: number, offset: number = 0): Promise<MemoryReadResult> {
+    async readMemory(location: string, length: number, offset = 0): Promise<Interfaces.MemoryReadResult> {
         const session = this.debugSessionManager.currentSession;
         if (session === undefined) {
             throw new Error('No active debug session.');
         }
 
         const result = await session.sendCustomRequest('cdt-gdb-adapter/Memory', {
             address: location,
-            length: length,
+            length,
             offset,
         });
 
         const bytes = hex2bytes(result.body.data);
-        const address = parseInt(result.body.address, 16);
+        const address = hexStrToUnsignedLong(result.body.address);
 
         return { bytes, address };
     }
+
+    // This functionality is not presently available from the CDT-GDB adapter
+    // async writeMemory(address: string, content: string): Promise<void> {
+    //     const { currentSession } = this.debugSessionManager;
+    //     if (!currentSession) {
+    //         throw new Error('No active debug session.');
+    //     }
+
+    //     await currentSession.sendCustomRequest('cdt-gdb-adapter/MemoryWrite', { address, content });
+    // }
```

Comment:
The write request is now in the DAP, so it should be possible to get it into the adapter.